src/DblParser/Main.ml

```diff
@@ -35,14 +43,20 @@ and repl_seq_main imported () =
 
   | Raw.REPL_Expr e ->
     let def = make_nowhere (Lang.Surface.DReplExpr(Desugar.tr_expr e)) in
+    let str = Bytes.to_string @@ Buffer.to_bytes buffer in
+    InterpLib.Error.repl_input := str;
```

Comment:
That doesn't seem to be the case since new buffer is created on every input, but its true that it can be created once and cleared before taking in any input.

Reply:
That's true. At first, I didn't understand the code correctly. A new buffer is created for each input. I prefer the previous version, with local buffers instead of the global one.